python/tank/descriptor/bundle_cache_usage/purger.py

```diff
@@ -0,0 +1,297 @@
+                    )
+
+            else:
+                raise BundleCacheUsageFileDeletionError(
+                    "Not a link, not a file, not a directory ??? : %s" % (f)
+                )
+
+    ###################################################################################################################
+    #
+    # PUBLIC API - methods & properties
+    #
+    # Can be called from any threading contextes (main or worker)
+    #
+    ###################################################################################################################
+
+    @property
+    def bundle_cache_root(self):
+        """
+        Returns the path to the specified bundle cache root folder.
+
+        :return: A str path, typically to the global bundle cache folder.
+        """
+        return self._database.bundle_cache_root
+
+    @LogManager.log_timing
+    def initial_populate(self):
+        """
+        Performs the initial-one-time search for bundles in the bundle cache and populate
+        the database as unused entries.
+        """
+        log.info("Searching for existing bundles ...")
+        found_bundles = self._find_bundles()
+        for bundle_path in found_bundles:
+            self._database.add_unused_bundle(bundle_path)
+
+        log.info("populating done, found %d entries" % (len(found_bundles)))
+
+        self._database.log_usage(self._marker_name)
+
+    @property
+    def initial_populate_performed(self):
+        """
+        :return: bool True if the initial database population was performed else False
+        """
+        return self._database.get_bundle(self._marker_name) is not None
+
+    def get_bundle_count(self):
+        """
+        Returns the number of currently tracked bundles in the database
+
+        :return: An int count
+        """
+        count = self._database.get_bundle_count()
+        # Subtract marker if present.
+        if self.initial_populate_performed:
+            count -= 1
+
+        return count
+
+    def get_unused_bundles(self, since_days=60):
+        """
+        Returns the list of bundles unused in the specified number of days.
+
+        :param since_days: an int count of days
+        :return: A list of :class:`~BundleCacheUsageDatabaseEntry`
+        """
+        oldest_timestamp = self._database._get_timestamp() - (since_days * 24 * 3600)
+        bundle_list = self._database.get_unused_bundles(oldest_timestamp)
+        # Remove marker entry if present
+        if self.initial_populate_performed:
+            for bundle in bundle_list:
+                if bundle.path.endswith(self.INITIAL_DB_POPULATE_DONE_MARKER):
+                    bundle_list.remove(bundle)
+
+        return bundle_list
+
+    def purge_bundle(self, bundle):
+        """
+        Delete both files and database entry relating to the specified bundle.
+
+        :param bundle: a :class:`~BundleCacheUsageDatabaseEntry` instance
+        """
+        try:
```

Comment:
I am sorry if i have perhaps confused things, i think for the deletion, it's sufficient if we execute http://developer.shotgunsoftware.com/tk-core/utils.html#sgtk.util.filesystem.safe_delete_folder - this method does not follow symlinks and will log and skip in the case of failures.